File: NG_manager_web/src/main/webapp/js/service/menuClassService.js
app.service('menuClassService',function ($http) {
    this.search=function (entity,page,rows) {
        return $http.post('../typetemplate/findPage.do?page='+page+"&rows="+rows,entity);
    };

    this.add=function (entity) {
        return $http.post('../typetemplate/add.do',entity);
    };

    this.findOne=function (id) {
        return $http.get('../typetemplate/findOne.do?id='+id);
    };

    this.update=function (entity) {
        return $http.post('../typetemplate/update.do',entity);
    };

    this.delete=function (ids) {
        return $http.get('../typetemplate/delete.do?ids='+ids);
    }

});

File: NG_dishes_interface/src/main/java/com/cqm/dishes/service/MenuClassService.java
package com.cqm.dishes.service;

import com.cqm.entity.PageResult;
import com.cqm.pojo.TbMenuClass;

import java.util.List;
import java.util.Map;

public interface MenuClassService {
    /**
     * 根据条件查询所有模板信息
     * @param menuClass
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbMenuClass menuClass,int pageNum,int pageSize);


    /**
     * 保存新增的模板对象
     * @param menuClass
     */
    public void add(TbMenuClass menuClass);

    /**
     * 根据id查找对应的模板对象
     * @param id
     * @return
     */
    public TbMenuClass findOne(Long id);

    /**
     * 根据id修改模板
     * @param menuClass
     */
    public void update(TbMenuClass menuClass);

    /**
     * 根据选中的id删除模板
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 查询所有的模板列表，为添加菜名提供数据
     * @return
     */
    public List<TbMenuClass> findAll();

}
